pulsar-broker/src/main/java/org/apache/pulsar/compaction/CompactorMXBeanImpl.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.compaction;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.LongAdder;
+
+public class CompactorMXBeanImpl implements CompactorMXBean {
+
+    private final ConcurrentHashMap<String, LongAdder> topicRemovedEventOps = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, Long> lastCompactSucceedTimestampOps = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, Long> lastCompactFailedTimestampOps = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, Long> lastCompactDurationTimeOps = new ConcurrentHashMap<>();
```

Comment:
How do we clean up the cache if the topic does not own by this broker.

Reply:
How about iterating the broker-owned topics with a fix delay to remove these topics?